File: Homework/themoviedb/src/main/java/com/example/themoviedb/persistence/LocalMovieDataSource.java
package com.example.themoviedb.persistence;

import com.example.themoviedb.MovieDataSource;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;

public class LocalMovieDataSource implements MovieDataSource {

    private final MovieDao mMovieDao;

    public LocalMovieDataSource(MovieDao movieDao) {
        mMovieDao = movieDao;
    }

    @Override
    public Flowable<Movie> getMovie(String title) {
        return mMovieDao.getMovie(title);
    }

    @Override
    public Flowable<List<Movie>> getAllMovies(){
        return mMovieDao.getAllMovies();
    }

    @Override
    public Flowable<List<Movie>> getWatchedMovies(){
        return mMovieDao.getWatchedMovies();
    }

    @Override
    public Completable insertOrUpdateMovie(Movie movie) {
        return mMovieDao.insertMovie(movie);
    }

    @Override
    public Completable deleteMovie(Movie movie) {
        return mMovieDao.deleteMovie(movie);
    }
}

File: Homework/themoviedb/src/main/java/com/example/themoviedb/persistence/LocalMovieDataSource.kt
package com.example.themoviedb.persistence

import com.example.themoviedb.MovieDataSource
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Maybe

class LocalMovieDataSource(private val mMovieDao: MovieDao) : MovieDataSource {

    override val allMovies: Flowable<List<Movie>>
        get() = mMovieDao.allMovies

    override val watchedMovies: Flowable<List<Movie>>
        get() = mMovieDao.watchedMovies

    override fun getMovie(title: String?): Maybe<Movie> {
        return mMovieDao.getMovie(title)
    }

    override fun insertMovie(movie: Movie): Completable {
        return mMovieDao.insertMovie(movie)
    }

    override fun updateMovie(movie: Movie): Completable {
        return mMovieDao.updateMovie(movie)
    }

    override fun deleteMovie(movie: Movie): Int {
        return mMovieDao.deleteMovie(movie)
    }
}